### ISSUE TITLE
Teleportation falsepositives

### ISSUE THREAD

**Issue by LogGits** [Zrips/CMI#1096]
**Description of issue or feature request:**
Had some false positives with the new teleportation system, but all in all its looking good so far.
A /tpahere while standing inside there, causes this:
https://gyazo.com/05542d3e134f96e5d0ba0d3c8efc40a5

A /tpahere while standing on soil under block causes this:
https://gyazo.com/8aa6191ee4c7624087e8f93fb9e7b568

I will keep updating it as i go 🙂 

---

**Cmi Version (using`/cmi version`):** 8.2.0.0

**Server Type (Spigot/Paperspigot/etc):** paperspigot

**Server Version (using `/ver`):**  build 456 1.13.2

**Comment by LogGits** [comment]
Ok so for anyone having this issue. Please change your config value of this to true `SafeLocationDownThenUp: false`
This fixes the issue:
`SafeLocationDownThenUp: true`

**Comment by mercurialmusic** [comment]
Definitely having the issue. The fix helps some, but not much. Seems like any warp or teleport point that doesn't have 3 blocks of clearance is sending players to the top of whatever, or failing without error.

**Comment by LogGits** [comment]
@mercurialmusic Can you please provide a screenshot of an example so we can test. Also Im not too sure if you said you tested it with the setting i wrote about but if not can you test with `SafeLocationDownThenUp` as true and then using `/cmi reload`. After that give it a test and let me know if its still happening.

**Comment by mercurialmusic** [comment]
@LogGits I had actually already switched to `SafeLocationDownThenUp: true` based on the changelog, hoping that would help, then came here when it still wasn't working and saw your suggestion to do the same. The switch did help some, allowing some players to access homes they weren't able to before I changed the setting, but the issue of any homes or warps in a 2 block tall space persists. I never had it happen for me so there may be some bypass permission coming into play, but the vast majority of my players were complaining rather loudly about it. 

It's most notably at our enderman farm, which is two blocks tall with barriers as the 3rd block and water on top of that. Every player says there's no error given, and I can confirm that there's no error in console. It fails completely silently even with a warp. They're able to use /res tp endgrinder from Residence in the meantime, but it's very annoying.

**Comment by Drc-DEV** [comment]
Would be useful a config option to disable safety entirely for certain TeleportTypes

**Comment by mercurialmusic** [comment]
> Would be useful a config option to disable safety entirely for certain TeleportTypes
> 
> The current teleport safety is totally broken. as of 8.2.0.0

Completely agreed. Right now I'm having to add the bypass perms to all my players just to restore normal functionality.

**Comment by DroidVengeance** [comment]
Also noticed if the player teleports from a world with the default floor height to a world with a lower floor height, it cancels and tells them "the teleport location is not safe void". This happens when using CMI portals, no issues until the latest update 8.2.0.0

**Comment by Zrips** [comment]
@DroidVengeance 
What is that default floor height? And do you have SafeLocationDownThenUp set to true or false?

**Comment by Zrips** [comment]
And for everyone else, todays update should solve issue with location detection while checking up from original position.

**Comment by mercurialmusic** [comment]
Nope, it's still doing it for me in 8.2.0.1.

**Comment by Blueeyestar** [comment]
The problem is still there with us despite the new version

**Comment by LogGits** [comment]
Can we get some server /version outputs, so we can get an idea as to what version we are working with. We just need more info. Pics false positives. Or if you could provide an ip i could come test the issue.

**Comment by Blueeyestar** [comment]
when do you want to test this?

**Comment by LogGits** [comment]
Right now would be perfect 🙂

**Comment by Blueeyestar** [comment]
if you have discord then you can show and explain it faster

**Comment by LogGits** [comment]
@Blueeyestar what is ur discord name il add you

**Comment by DroidVengeance** [comment]
@Zrips  Sorry, I didn't get this notification, the main world's floor height is 65, the world they are teleporting into is 11. I've tried SafeLocationDownThenUp TRUE and FALSE with the same issue. I'll update to 8.2.0.1 right now and let you know if it still occurs.

**Comment by DroidVengeance** [comment]
@Zrips The message still shows with the latest update. I tried giving the unsafetp permission as well with no luck.
![ss_ 2018-11-27 08 02 56](https://user-images.githubusercontent.com/13245547/49083554-f0556580-f21a-11e8-9f78-7373c1a587cb.png)

**Comment by mercurialmusic** [comment]
@LogGits The one's I've seen are cross-world, yes, but it doesn't seem to matter what type of world. I'm sorry I can't provide screen shots since for us there's nothing to show. Simply nothing happens, or in the case of our enderman farm that I described, they're put on top of it (even with `SafeLocationDownThenUp: true`). As a temporary fix we moved the end warp to a section that had a 2.5-3 block clearance and it works for now. For the homes that weren't working that head clearance also seems to have been the fault (clearing out an additional block allows the home to work again), so it seems like there's something wrong in the check for how much room is allowed?

CMI plugin version: 8.2.0.1
Server version: Spigot 1.13.2-R0.1-SNAPSHOT
Vault version: 1.6.11(+)
